platforms/native/language-native/src/main/java/org/gradle/language/internal/DefaultBinaryCollection.java

```diff
@@ -194,6 +194,7 @@ public Class<S> getType() {
         }
 
         // Mix in some Groovy DSL support. Should decorate instead
+        @SuppressWarnings("unused") // DI configuration
```

Comment:
might undo this again as the issue remains:

- https://github.com/gradle/gradle/pull/35470#issuecomment-3447380237

```suggestion
        //@SuppressWarnings("unused") // DI configuration
```